api/command_join.go

```diff
@@ -45,7 +45,7 @@ func (me *JoinProvider) DoCommand(c *Context, channelId string, message string)
 				if v.Type != model.CHANNEL_OPEN {
 					return &model.CommandResponse{Text: c.T("api.command_join.fail.app_error"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
 				}
-				return &model.CommandResponse{GotoLocation: c.GetTeamURL() + "/channels/" + v.Name, Text: c.T("api.command_join.success"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
+				return &model.CommandResponse{GotoLocation: v.Name + "/channels/" + v.Id, Text: c.T("api.command_join.success"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
```

Comment:
That is intentional, otherwise client side handling becomes much more difficult (asynchronous calls to `getMoreChannels()`)